### PR TITLE
Simplify getting cache dir to support other POSIX-like systems

### LINKED ISSUE #49: Patch Path.h to allow for external definition of UTILS_OS_POSIX, UTILS_OS_LINUX, ...
Hello! I'm working on compiling the [webthree-umbrella](https://github.com/ethereum/webthree-umbrella/) source tree (WU) on FreeBSD 10.3-RELEASE. WU uses [libethereum](https://github.com/ethereum/libethereum) as a Git submodule, which of course - in turn - uses evmjit as a submodule. 

Personally, I'd like to be able to develop a complete port and some documentation about the WU build on FreeBSD - perhaps towards developing a manner of a FreeBSD edition specifically for Ethereum nodes? and of course it would include `geth`. Presently, I'm trying to - so to speak - to "Hash out" the dependencies for the build, onto the FreeBSD ports tree.

I noticed that `libevmjit/support/Path.h` defines a small number of preprocessor macros, such that are applied in the corresponding `Path.cpp` file as with regards to the definitions of the `dev::path::user_home_directory()` and `dev::path::user_cache_directory()` functions. Rather than trying to globally define `-D__linux` to pull in those macros' respective Linux definitions on FreeBSD, I've patched `Path.h` such as to allow for the macros to be declared in `CXXFLAGS`. The macros affected with this patch include:

- **`UTILS_OS_LINUX`**

-  `UTILS_OS_MAC`

- **`UTILS_OS_POSIX`**

- `UTILS_OS_WINDOWS`

The applications of the **`bold`** annotated macros, of course, may be in some ways apropos to the FreeBSD build.

In reference to how these macros are applied in `Path.cpp` I've added a note to my own albeit ad-hoc notebook - towards adding a note to the documentation - that the  `UTILS_OS_LINUX` definition is apropos to an environment variable named **`XDS_CACHE_HOME `** applied in `Path.cpp`. I'm not sure of whether or how that may pertain to `XDG_CACHE_HOME` as in reference to the [XDG Base Directory](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) spec. It seems that the `XDS...` environment variable's name does find some applications, however I'm not sure if it's quite as common as the `XDG_CACHE_HOME` form? Perhaps that could be addressed in a separate patch, but I wouldn't want to propose a change to it without being certain as with regards to applications of the `XDS` form _viz a viz_ `XDG_CACHE_HOME`

The present patch will address only the definitions in `Path.h`

[patch_evmjit_path.diff.txt](https://github.com/ethereum/evmjit/files/241052/patch_evmjit_path.diff.txt)

### PR DESCRIPTION
**Pull request by chfast**
Resolves #49.

### CODE REVIEW

File: libevmjit/Cache.cpp [hunk 1]
```diff
@@ -33,7 +33,7 @@ namespace
 	std::string getVersionedCacheDir()
 	{
 		llvm::SmallString<256> path{path::user_cache_directory()};
-		static const auto c_ethereumAppName = UTILS_OS_LINUX ? "ethereum" : "Ethereum";
+		static const auto c_ethereumAppName = UTILS_OS_WINDOWS || UTILS_OS_MAC ? "Ethereum" : "ethereum";
```
Comment (chriseth) [inline libevmjit/Cache.cpp]: Parentheses for disambiguation?

**Comment by chriseth** [comment]
*thumbsup*

**Comment by bobsummerwill** [comment]
Looks good to me too.

WRT to the "Ethereum" versus "ethereum" difference, I have seen some warts like that in the build system too, ie.

https://github.com/ethereum/webthree-helpers/blob/develop/cmake/EthExecutableHelper.cmake#L173

We really shouldn't need these, should we?      I repeatedly saw this kind of anti-pattern at EA where development on a codebase had started on Windows, so we had some inconsistencies in filenames in version control, or with property names in build files, etc.     Then we would start supporting Linux, of have a Perforce server running on Linux or whatever, and they would bite us.

I'm just wondering if you guys know the origins of these inconsistencies, and whether or not we can just fix them and remove the warts?

For this cache example, OS X and Windows are both case-insensitive, so searching for "ethereum", same as Windows, should actually work too.

**Comment by chfast** [comment]
@bobsummerwill I agree it is bad pattern. I'm guessing I wanted to match other application from C++ team.

This naming issue is more about matching OS-specific naming convention. E.g. the executable of AlethZero is named AlethZero on MacOS and alethzero on Linux.

I believe we should select one name and stick to it no matter what OS. I was very close to using "Ethereum" for dir name in this PR, but it is better not to mix functional changes.

**Comment by bobsummerwill** [comment]
Well, what we certainly could do is stick consistently to lowercase for all platforms.

That change of case would not break anything on OS X or Windows, where changing the case on Linux would break things.

Many teams at EA actually had a standard along that line, for ease of portability - "stick to lowercase for directories and filenames".

**Comment by bobsummerwill** [comment]
Anyway - merge away.   We can talk about casing separately.

**Pull request merged**